src/hiprtc.cpp

```diff
@@ -301,12 +301,12 @@ struct _hiprtcProgram {
     }
 
     void replaceExtension(std::string& fileName, const std::string &ext) const {
-        auto res = std::find(fileName.rbegin(), fileName.rend(), '.');
-        auto sloc = std::find(fileName.rbegin(), fileName.rend(), '/');
-        if(res == fileName.rend() || res > sloc) {
-            fileName += ext;
+        auto res = fileName.rfind('.');
+        auto sloc = fileName.rfind('/');
+        if (res != std::string::npos && res > sloc) {
```

Comment:
This needs to catch the case where `/` is not found, the condition should be
`if (res != npos && (res > sloc || sloc == npos))`

Reply:
if `/` is not found, `sloc `should have the maximum possible value for `size_t `and the condition will fail.

Reply:
if '.' is found but '/' is not found, don't you want to go to this path as well?

Reply:
fixed it.